app/models/display_ad.rb

```diff
@@ -187,4 +189,22 @@ def should_refresh_audience_segment?
       audience_segment &&
       audience_segment.updated_at < 1.day.ago
   end
+
+  def valid_audience_segment_type
+    return if audience_segment_type.blank?
+
+    errors.add(:audience_segment_type) unless AudienceSegment.valid_type?(audience_segment_type)
+  end
```

Comment:
I'm fairly sure it's possible to express this as an `inclusion` validation, which would have Rails provide a nice error message

(Also would be nice to have readable error messages for the other validations)